internal/core/textlayout/sharedparley.rs

```diff
@@ -0,0 +1,142 @@
+// Copyright © SixtyFPS GmbH <info@slint.dev>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-2.0 OR LicenseRef-Slint-Software-3.0
+
+pub use parley;
+
+use std::cell::RefCell;
+
+use crate::{
+    graphics::FontRequest,
+    items::TextStrokeStyle,
+    lengths::{LogicalLength, PhysicalPx},
+    textlayout::{TextHorizontalAlignment, TextOverflow, TextVerticalAlignment, TextWrap},
+};
+use i_slint_common::sharedfontique;
+
+pub const DEFAULT_FONT_SIZE: LogicalLength = LogicalLength::new(12.);
+
+fn font_context() -> parley::FontContext {
+    parley::FontContext {
+        collection: sharedfontique::COLLECTION.inner.clone(),
+        source_cache: sharedfontique::COLLECTION.source_cache.clone(),
+    }
+}
```

Comment:
Even with the shared `FontContext` you probably don't want to be cloning it for every layout. I would recommend using a thread_local for this too. You'd probably get away with the clones, but you would be leaving performance on the table.

(improving this design is something we'll hopfully be looking at over the next few months)